includes/components/class-attribute.php

```diff
@@ -91,6 +94,25 @@ protected function boot() {
 		}
 	}
 
+	/**
+	 * Inits models properties.
+	 */
+	public function init_models_properties() {
+
+		// Get post types.
+		$post_types = hivepress()->get_config( 'post_types' );
+
+		foreach ( $this->get_models() as $model ) {
+
+			// Set defaults.
+			$this->models[ $model ]['searchable'] = true;
+
+			if ( ! hp\get_array_value( hp\get_array_value( $post_types, $model, [] ), 'has_archive' ) ) {
+				$this->models[ $model ]['searchable'] = false;
```

Comment:
Please ensure that moving this flag initialization later doesn't affect all the places where this flag is used (search for all the "searchable" occurances), e.g. maybe meta boxes on the back-end are affected (for example check the booking attribute page, there should be no Search meta box settings).

Also, please exclude user from the list of post type checks to prevent a future conflict (e.g. if someone registers the post type with the "user" name, there's a possibility).